homeassistant/components/mysensors.py

```diff
@@ -0,0 +1,193 @@
+"""
+homeassistant.components.mysensors
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+MySensors component that connects to a MySensors gateway via pymysensors
+API.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.mysensors.html
+
+
+New features:
+
+New MySensors component.
+Updated MySensors Sensor platform.
+New MySensors Switch platform.
+Multiple gateways are now supported.
+
+Configuration.yaml:
+
+mysensors:
+  gateways:
+    - port: '/dev/ttyUSB0'
+      persistence_file: 'path/mysensors.json'
+    - port: '/dev/ttyACM1'
+      persistence_file: 'path/mysensors2.json'
+  debug: true
+  persistence: true
+  version: '1.5'
+"""
+import logging
+
+from homeassistant.helpers import validate_config
+import homeassistant.bootstrap as bootstrap
+
+from homeassistant.const import (
+    EVENT_HOMEASSISTANT_START,
+    EVENT_HOMEASSISTANT_STOP,
+    TEMP_CELCIUS,
+    CONF_PLATFORM)
+
+CONF_GATEWAYS = 'gateways'
+CONF_PORT = 'port'
+CONF_DEBUG = 'debug'
+CONF_PERSISTENCE = 'persistence'
+CONF_PERSISTENCE_FILE = 'persistence_file'
+CONF_VERSION = 'version'
+DEFAULT_VERSION = '1.4'
+VERSION = None
+
+DOMAIN = 'mysensors'
+DEPENDENCIES = []
+REQUIREMENTS = [
+    'https://github.com/theolind/pymysensors/archive/'
+    '2aa8f32908e8c5bb3e5c77c5851db778f8635792.zip#pymysensors==0.3']
+_LOGGER = logging.getLogger(__name__)
+ATTR_NODE_ID = 'node_id'
+ATTR_CHILD_ID = 'child_id'
+
+COMPONENTS_WITH_MYSENSORS_PLATFORM = [
+    'sensor',
+    'switch',
+]
+
+IS_METRIC = None
+CONST = None
+GATEWAYS = None
+
+
+def setup(hass, config):
+    """Setup the MySensors component."""
+    # pylint: disable=too-many-locals
+    import mysensors.mysensors as mysensors
+
+    if not validate_config(config,
+                           {DOMAIN: [CONF_GATEWAYS]},
+                           _LOGGER):
+        return False
+
+    global VERSION
+    VERSION = config[DOMAIN].get(CONF_VERSION, DEFAULT_VERSION)
+
+    global CONST
+    if VERSION == '1.5':
```

Comment:
YAML is 'smart' and will convert your version to a number unless enclosed in quotes. To prevent confusion for people, let's make sure you cast the config to a string.